pubtools/pulplib/_impl/client/client.py

```diff
@@ -181,10 +184,87 @@ def search_repository(self, criteria=None):
             response_f, lambda data: self._handle_page(Repository, search, data)
         )
 
+    def get_maintenance_report(self):
+        """Get maintenance report from maintenance repository
+
+        Returns:
+            :class:`~pubtools.pulplib.MaintenceReport` object describes the
+            maintenance status
+
+            None if there's no report in the repository
+        """
+        try:
+            data = self._do_get_maintenance().result()
+        except requests.exceptions.HTTPError:
+            # if the maintenance mode hasn't been used before, then it could return
+            # HTTPError 404.
+            LOG.warning("No maintenance info found")
+            return None
```

Comment:
I wouldn't recommend returning None, isn't it more user-friendly to return a maintenance report which shows that no repos are in maintenance mode?

If someone is writing code dealing with maintenance mode, is there any advantage in forcing them to write separate code for the None case vs "nothing is in maintenance mode" case?